avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,311 @@
+import logging
+import multiprocessing
+import os
+import time
+import traceback
+from abc import ABC, abstractmethod
+
+from avocado.core import nrunner
+from avocado.core.runners.utils import messages
+from avocado.core.settings import settings
+from avocado.utils import astring, process, software_manager
+
+log = logging.getLogger("avocado.sysinfo")
+
+
+class _Collectible(ABC):
+
+    """
+    Abstract class for representing collectibles by sysinfo.
+    """
+
+    def __init__(self, logf):
+        self.logf = astring.string_to_safe_path(logf)
+
+    @abstractmethod
+    def run(self, queue, logdir):
+        pass
+
+
+class _Logfile(_Collectible):
+
+    """
+    Collectible system file.
+
+    :param path: Path to the log file.
+    """
+
+    def __init__(self, path):
+        super().__init__(os.path.basename(path))
+        self.path = path
+
+    def __repr__(self):
+        r = "sysinfo.Logfile(%r, %r)"
+        r %= (self.path, self.logf)
+        return r
+
+    def __eq__(self, other):
+        if isinstance(other, _Logfile):
+            return (self.path, self.logf) == (other.path, other.logf)
```

Comment:
Here, maybe, you could do something like:

```python
return hash(self) == hash(other)
```

In that way, we could remove the duplicated  codes, so  each class just need to re-implement  the   `def __hash__()` methods.